### PR TITLE
Return a correct exit code when using the command line utility to check certificates and include the ip/email/host validation return code

### PR DESCRIPTION
**Pull request by ramikhaldi**
Hi,

This merge request aims to fix the issue with the returned exit code 0, even if there are validation errors and/or something has failed to perform correctly while executing the openssl command line utility. I intentionally avoided a big refactoring to make the commit as clean and short as possible and to perform a fix for now. I appreciate each feedback :)

KR,
Rami

### CODE REVIEW

File: apps/lib/apps.c
Comment (FdaSilvaYY) [inline apps/lib/apps.c:2119]: nit: project style is 4-spaces indentation

File: apps/x509.c
Comment (FdaSilvaYY) [inline apps/x509.c:1041]: nit: bad indentation

**Review by FdaSilvaYY — COMMENTED** [review]
No merge commit is allowed here.
some indentation issues.

File: apps/lib/apps.c
Comment (ramikhaldi) [reply apps/lib/apps.c]: fixed in last commit.

File: apps/x509.c
Comment (ramikhaldi) [reply apps/x509.c]: fixed in last commit.

**Comment by t8m** [comment]
The pull request may not contain any merge commits. You can get rid of it by using `git rebase -i`, dropping the merge commit and force pushing it.

Also this is definitely not within the CLA: trivial limits, so please remove the CLA: trivial from the commit messages (you can do that via the `git rebase -i` as well.
We will need a regular CLA submitted. https://www.openssl.org/policies/cla.html

**Comment by mattcaswell** [comment]
Close/reopen to kick CLA bot.

**Comment by mattcaswell** [comment]
Hmmm. CLA bot did not remove the hold as expected. The entry in the cladb looks correct. I suspect this is a symptom of the ongoing repo "mirroring" problems we've been having. I've manually removed the hold for now. Lets hope the CLA bot doesn't decide to put it back.

**Comment by ramikhaldi** [comment]
I reverted a changed behavior regarding the exitcode in case of noout or nocert. According to the tests, it seems, that someone wanted to have a 0 exit code in these cases. So, the tests should run green now.

**Comment by ramikhaldi** [comment]
@mattcaswell the bot added the label again.

File: apps/lib/apps.c
Comment (paulidale) [inline apps/lib/apps.c:2120]: Coding style nit: blank line between declarations and code.

Here and below.

File: apps/x509.c
Comment (paulidale) [inline apps/x509.c:1068]: We more typically use `err` for this king of thing.  Not important though.

File: apps/include/apps.h
Comment (t8m) [inline apps/include/apps.h:266]: I would not use X509_.... name as those are usually public libcrypto API calls.

**Review by paulidale — APPROVED** [review]

**Comment by mattcaswell** [comment]
> Could you please remove it in order to trigger the ci again ?

The bot fixed itself, and I just approved the CI run.

File: apps/include/apps.h
Comment (t8m) [inline apps/include/apps.h:266]: I still do not like the name of the function as this one does not really describe what the function does. Perhaps `check_cert_data` or something like that?

File: apps/lib/apps.c
Comment (t8m) [inline apps/lib/apps.c:2121]: When you're touching the code, could you please change all the occurrences of pointer to boolean conversion in the function to `pointer != NULL` ?

File: apps/lib/apps.c
Comment (t8m) [inline apps/lib/apps.c:2124]: Could you please split the overly long line?

File: apps/lib/apps.c
Comment (t8m) [inline apps/lib/apps.c:2131]: Please split the line

File: apps/lib/apps.c
Comment (t8m) [inline apps/lib/apps.c:2138]: As above

File: apps/x509.c
Comment (t8m) [inline apps/x509.c:1069]: Please fix the indentation.

File: apps/x509.c
Comment (t8m) [inline apps/x509.c:1040]: There should be no space after `!`

**Review by t8m — CHANGES_REQUESTED** [review]

File: apps/include/apps.h
Comment (ramikhaldi) [reply apps/include/apps.h:266]: As these data are the certificate attributes, I named the function as `check_cert_attributes`

File: apps/lib/apps.c
Comment (t8m) [inline apps/lib/apps.c:2125]: I am sorry but this line and the similar lines below need to be indented so checkhost is placed after the `(` on the line above.

**Review by t8m — APPROVED** [review]

**Comment by ramikhaldi** [comment]
Can we merge it to the 3.0 branch as well ?

**Comment by paulidale** [comment]
Merging to 3.0 and 1.1.1 would require OTC approval I think.  This is fixing a problem (bad exit code) which I'd consider to be a bug rather than a feature.

I'll raise it for discussion by the OTC.

**Comment by paulidale** [comment]
Question for OTC: **is this a bug fix and thus suitable for merging to 3.0 and 1.1.1 releases?**

The x509 command is returning a successful exit when problems are detected in the host, email or IP addresses in the cert.  This change increases the level of checking and returns an error from the command if there are problems detected.  I.e. it is a breaking change.

**Review by paulidale — APPROVED** [review]

**Comment by rsbeckerca** [comment]
Could this possibly be related to #16950, which works on 3.0, 1.1, but not master?

**Comment by paulidale** [comment]
No, I don't think it's related.  The error code is good, just wrong.

**Comment by ramikhaldi** [comment]
Does it mean, that due to the new comments after label „approved“ has been set, the OpenSSL bot could not automatically set the label „read to merge“ ?

**Comment by paulidale** [comment]
That's what it means.  I'll just merge this.

**Comment by ramikhaldi** [comment]
I accidentally purged my master branch. I am working on fixing it ... sorry for the inconvenience.

**Comment by ramikhaldi** [comment]
I fixed the PR.

File: apps/include/apps.h
Comment (paulidale) [inline apps/include/apps.h:345]: New line here please.

File: apps/lib/apps.c
Comment (paulidale) [inline apps/lib/apps.c:3357]: new line here too please.

File: apps/x509.c
Comment (paulidale) [inline apps/x509.c:1267]: new line here three please.

**Review by paulidale — CHANGES_REQUESTED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Squashed, reworded the commit message and merged to master. Thank you!

Keeping this open for eventual cherry-pick if OTC approves.

**Review by paulidale — APPROVED** [review]
Needs the formality of an approve even if it is post hoc.

**Comment by mattcaswell** [comment]
OTC: not a bug fix

Closing since already merged to master.

**Pull request closed without merging**
